File: BookingAppServer/ApplicationServices/Models/Menu/DishVm.cs
using ApplicationServices.Models.Enums;

namespace ApplicationServices.Models.Menu;

public class DishVm
{
    public string Id { get; set; }

    public string Category { get; set; } = string.Empty;

    public string Name { get; set; } = string.Empty;

    public string Description { get; set; } = string.Empty;

    public int Calories { get; set; } = 0;

    public double Price { get; set; }

    public string Image { get; set; } = string.Empty;

    public DateTime CreatedAt { get; set; }

    public DateTime? ModifiedAt { get; set; }
}

File: BookingFood/BookingFood/Models/DishVm.swift
//
//  DishVm.swift
//  BookingFood
//
//  Created by Huy Binh on 30/08/2022.
//

import Foundation

struct DishVm : Codable {
    
    let id,name: String
    let description,category,image: String?
    let calories: Int?
    let price: Double
    
    var cartId: String?
    
    var formattedCalories: String {
        return "\(calories ?? 0) calories"
    }
    
    var formattedPrice: String {
        return price.toCurrency()
    }
    
}
